unlock_indicator.c

```diff
@@ -678,6 +685,19 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
     cairo_surface_t *xcb_output = cairo_xcb_surface_create(conn, bg_pixmap, vistype, resolution[0], resolution[1]);
     cairo_t *xcb_ctx = cairo_create(xcb_output);
 
+    /*update image according to the slideshow_interval*/
+    if (slideshow_image_count > 0) {
+        unsigned long now = (unsigned long)time(NULL);
+        if (-1 == lastCheck || now - lastCheck >= slideshow_interval) {
```

Comment:
`((lastCheck == -1) || (now - lastcheck >= slideshow_interval))`

realistically I think that `lastCheck` could be set somewhere - maybe in `init_colors_once` or something? I can't think of an ideal place to set it off the top of my head.